File: lib/network/packets/server/s_delete_char_result.dart
class SDeleteCharResult {
  const SDeleteCharResult({
    required this.success,
    required this.reason,
    required this.message,
  });

  final bool success;
  final String reason;
  final String message;

  factory SDeleteCharResult.fromData(Map<String, dynamic> data) {
    return SDeleteCharResult(
      success: data['success'] as bool? ?? false,
      reason: data['reason'] as String? ?? '',
      message: data['message'] as String? ?? '',
    );
  }
}

File: src/main/java/com/xin/server/packet/server/S_DeleteCharResult.java
package com.xin.server.packet.server;

import com.xin.server.packet.ServerBasePacket;
import com.xin.server.packet.ServerOpcodes;

public class S_DeleteCharResult extends ServerBasePacket {

    public static final String REASON_OK = "OK";
    public static final String REASON_NOT_FOUND = "NOT_FOUND";
    public static final String REASON_DB_ERROR = "DB_ERROR";

    private S_DeleteCharResult(boolean success, String reason, String message) {
        super(ServerOpcodes.S_DELETE_CHAR_RESULT);
        put("success", success);
        put("reason", reason);
        put("message", message);
    }

    public static S_DeleteCharResult ok() {
        return new S_DeleteCharResult(true, REASON_OK, "角色刪除成功");
    }

    public static S_DeleteCharResult fail(String reason, String message) {
        return new S_DeleteCharResult(false, reason, message);
    }
}
